SAMD21_LPM

immesys opened the pull request:
Opening this pull request on behalf of LudwigKnuepfer to test my power baselining bot

Commit:
cpu/samd21: implement low power modes

Commit:
sys/rtt_stdio: Support SEGGER RTT for stdin/stdout

immesys commented:
@profilebot please recheck this

immesys commented:
@profilebot please recheck this, your formatting sucks

immesys commented:
Ok, I've enabled all the apps, @profilebot please recheck this?

immesys commented:
Seriously though. Nevermind artificial intelligence, @profilebot you are artificial stupidity.

immesys commented:
@profilebot please recheck this

immesys commented:
Ok, try it with increased accuracy: @profilebot please recheck this

immesys commented:
@profilebot please recheck this and use estimate mode

immesys commented:
@profilebot please recheck this and use precise mode. Might as well do something overnight.

PeterKietzmann commented:
If I see it correctly there is just data in the first plot

immesys commented:
Yeah, I fixed that bug. There is still a bug where it uses estimate precision even when you ask for precise mode. I'll fix that later. Right now though, @profilebot please recheck this and use estimate mode

immesys commented:
OK I guess I have not fixed the graph bug yet..

immesys commented:
@profilebot please recheck this and use precise mode

immesys commented:
@PeterKietzmann that is what the graphs are supposed to look like.

PeterKietzmann commented:
@immesys looks nice! The relations seem reasonable to me. How about absolute values, do you know where to find the specs in the datasheet?

aabadie committed:
boards/arduino-zero: add initial support

cgundogan committed:
gnrc_rpl: netstats_rpl: introduce statistics

cgundogan committed:
examples/gnrc_networking: use netstats_rpl

cgundogan committed:
gnrc_rpl: shell: add 'rpl stats' cmd to shell

OlegHahm committed:
native: enable valgrind's leak summary

astralien3000 committed:
Aversive++: initial import

Commit:
hashes/sha256: changed data pointer parameters to `void*`

Commit:
hashes/md5: changed data-parameter types to `void*`

miri64 committed:
gnrc_pktbuf: allow for 0-sized payload snips

miri64 committed:
tests: adapt unittests for new gnrc_pktbuf preconditions

basilfx committed:
pkg: u8g2: upgrade package to latest version.

New version introduces better support for the different SPI modes.

basilfx committed:
tests: pkg_u8g2: simplify Makefile + default display per output type

This matches better with other tests and makes code less bloated.

miri64 committed:
od: don't dump 0-length chunks

aabadie committed:
cpu/stm32f4: adding support for stm32f446re cpu

aabadie committed:
boards/nucleo-f446: initial support

Commit:
cpu/samr21: rename ASF include files to prepare for ASF update

Commit:
cpu/sam21_common: update ASF CMSIS (samr21) include files using ASF 3.21

Notice that RIOT was using FDPLL96M, which is indeed the name in the
datasheet. But Atmel is using GCLK_GENCTRL_SRC_FDPLL.

Commit:
cpu/sam21_common: add a few typedefs for SERCOM

A generic function is added to initialize a SERCOM.

Notice that uart_conf_t was expanded with pad settings, but it isn't used
yet.

Commit:
cpu/sam21_common: move i2c.c from samd21 to sam21_common

Notice that saml21 does not use I2C (yet).

Commit:
cpu/samd21: rewrite i2c_init_master to make use of gpio_init_sercom

The former i2c_init_master was too complicated. It was trying to set IN/OUT
mode of the pins, but all that is needed is to set the proper MUX.

Also the configuration for the boards was incomplete (no MUX, no pad
setting).

It was tested on a SODAQ Autonomo, but not on a samr21-xpro

Commit:
cpu/samd21: spi: change #ifdef's to #if's

The board config can define SPI_1_EN as 0, and in that case #ifdef won't
work.

Add some more comments, and fix typos.

Commit:
cpu/samd21: Refactor spi using ArduinoCore as example

The pinmux is now part of the board config. The pad setting is done with
clear names instead of numbers.

Commit:
cpu/samd21: uart: now using padd setting from board config

Notice the extra fields in uart_conf_t for rx_pad and tx_pad.

kaspar030 committed:
cpu: cortexm_common: fix compile warning

Commit:
doc: correct exclude patterns after moving CMSIS files to cpu/sam21_common

Commit:
cpu/sam21_common: eliminate trailing white space in CMSIS include files

miri64 committed:
tests: provide tests for od module

Commit:
dist/tools: exclude sam21_common cmsis include files

Commit:
cpu/samd21: introduce sam0.h so that we can select samr21.h or other

In the Makefile.include of the board it is required to set the correct
define.

Currently only SAMRG18A is supported, as required by samr21-xpro.

Commit:
cpu/sam21_common: add samd21 include files from ASF CMSIS

This adds support for all samd21 variants. Notice that the previous samd21
was moved to cpu/samr21

To use this cpu variant you must set at least the following in your boards
Makefile.include:
    export CPU = samd21
    export CPU_MODEL = samd21j18a   # Not sure if this is used
    CFLAGS += -D__SAMD21J18A__  # or whatever variant you have

The samd21 include files were copied from ASF 3.30

Commit:
cpu/samd21: add helper functions spi_power_on/off

Commit:
cpu/sam21_common: add C++ wrapper to sam0.h

jthacker committed:
atmega: use software interrupt for context swap

Fixes #5745
For AVR based boards, three defines must be defined AVR_CONTEXT_SWAP_INIT,
AVR_CONTEXT_SWAP_INTERRUPT_VECT, and AVR_CONTEXT_SWAP_TRIGGER.
These defines are used to trigger a software interrupt used for context
switching.

When AVR_CONTEXT_SWAP_INTERRUPT_VECT is handled, the scheduler is run
and a context swap will happen if necessary, with the resulting thread
starting following the reti instruction. This results in threads running
at normal priority instead of at interrupt priority.

Atmega devices do provide a pure software interrupt. The method used
here for waspmote-pro and arduino-mega2560 is to use pin change
interrupts, set the pin to act as an output, and toggle the value to
simulate a software interrupt. The main limitation here is that a
physical pin is now occupied and must be defined for each board
supported by RIOT. On the plus side, it provides an easy method for
detecting context swaps with an oscilloscope.

makomi committed:
cpu/stm32f0: uart: make sure that 'port' and 'dev' are !=NULL

This fixes some cppcheck nullPointer errors.

makomi committed:
cpu/stm32f3: uart: make sure that 'port' and 'dev' are !=NULL

This fixes some cppcheck nullPointer errors.

makomi committed:
cpu/stm32l1: uart: make sure 'dev' is != NULL to avoid nullPointer errors

makomi committed:
pkg/nordic_softdevice_ble: gnrc_nordic_ble_6lowpan: fix nullPointer error

makomi committed:
pkg/emb6: emb6_conn_udp: fix nullPointer error

makomi committed:
sys/net/gnrc: gnrc_zep: fix nullPointer error

makomi committed:
dist/tools/tunslip: add comment regarding missing break at the end of case

makomi committed:
sys/base64: fix variableScope

makomi committed:
drivers/dht: fix variableScope

makomi committed:
sys/ecc/hamming256: fix variableScope

makomi committed:
sys/net/gnrc: gnrc_ipv6_ext: fix variableScope

makomi committed:
sys/net/gnrc: gnrc_rpl_control_messages: fix variableScope

makomi committed:
sys/shell: fix cppcheck warnings

makomi committed:
tests/driver_dht: fix invalidPrintfArgType_sint

Since the argument type is 'unsigned int', the printf
format string has to be '%u' instead of '%i'.

makomi committed:
tests/periph_uart: fix invalidPrintfArgType_sint

Since the argument type is 'unsigned int', the printf
format string has to be '%u' instead of '%i'.

makomi committed:
tests/periph_gpio: fix variableScope

makomi committed:
tests/xtimer_now64_continuity: fix variableScope

makomi committed:
dist/tools/tunslip: fix variableScope

makomi committed:
dist/tools/tunslip: tunslip6: fix variableScope

makomi committed:
dist/tools/tunslip: tunslip6: fix invalidPrintfArgType_uint

Since the argument type is 'long', the printf
format string has to be '%li' instead of '%lu'.

makomi committed:
dist/tools/tunslip: tapslip6: fix variableScope

makomi committed:
dist/tools/tunslip: tapslip6: fix unusedVariable

smlng committed:
native: correcting linker LTO flags for OSX

  - corrects and enhances changes introduced in #5742

miri64 committed:
dist: document IOTLAB_PROFILE variable fo IoT-Lab-support

miri64 committed:
at86rf2xx: document operation mode concerns for

miri64 committed:
fixup! sock: Introduction of new application layer API

Add more error classes

mali committed:
cpu/atmega328p: add atmega328p mcu support

add atmega328p support with:
uart, timer, spi and gpio

OlegHahm committed:
Merge pull request #5800 from miri64/at86rf2xx/doc/get_random_operation

at86rf2xx: document operation mode concerns for

miri64 committed:
auto_init_gnrc_netif: set priorities one lower than 6LoWPAN

aeneby committed:
cpu/cc2538: Remove superfluous mutex lock

OlegHahm committed:
Merge pull request #5653 from aeneby/rssi_lqi_fix_v2

cpu/cc2538: Make ED and LQI values 802.15.4 compliant

Commit:
Merge pull request #5806 from makomi/base64

sys/base64: fix variableScope (cppcheck)

miri64 committed:
Merge pull request #5808 from makomi/dht

drivers/dht: fix variableScope (cppcheck)

Commit:
vagrant: info about vboxusers group for USB access

miri64 committed:
Merge pull request #5811 from makomi/gnrc

sys/net/gnrc: fix cppcheck errors/ warnings

miri64 committed:
Merge pull request #5809 from makomi/driver_dht

tests/driver_dht: fix invalidPrintfArgType_sint (cppcheck)

miri64 committed:
Merge pull request #5810 from makomi/emb6

pkg/emb6: emb6_conn_udp: fix nullPointer error (cppcheck)

miri64 committed:
Merge pull request #5812 from makomi/periph_uart

tests/periph_uart: fix invalidPrintfArgType_sint (cppcheck)

kaspar030 committed:
Merge pull request #5820 from makomi/xtimer_now64_continuity

tests/xtimer_now64_continuity: fix variableScope (cppcheck)

kaspar030 committed:
Merge pull request #5814 from makomi/nordic_softdevice_ble

pkg/nordic_softdevice_ble: gnrc_nordic_ble_6lowpan: fix nullPointer error (cppcheck)

OlegHahm committed:
Merge pull request #5813 from makomi/hamming256

sys/ecc/hamming256: fix variableScope (cppcheck)

aabadie committed:
Merge pull request #5702 from basilfx/feature/pkg_u8g2_improvements

pkg: u8g2: improvements and bugfixes

mali committed:
boards: add arduino uno and duemilanove support

Uno and Duemilanove(atmega328p version) are nearly the same boards.
The only difference is that the Duemilanove use an FTDI usb chip,
while the Uno use an Atmel which acts as USB/Serial converter.
All of the code needed to support these boards is in arduino-common.

thomaseichinger committed:
dist/tools: include uname check for FreeBSD in genconfigheader

Analog to Darwin set `MD5SUM` to `md5 -r`.

Commit:
vagrant: update ubuntu version in readme

Commit:
vagrant: change RIOT-OS to RIOT in readme

miri64 committed:
make: build quiet as default

Commit:
packer: intial conf. to generate virtualbox based vagrant box

Commit:
vagrant: adujst Vagrantfile to use new RIOT-VM

smlng committed:
Merge pull request #5826 from thomaseichinger/pr/fix_freebsd

dist/tools: include uname check for FreeBSD in genconfigheader

Commit:
vagrant: restructure 'addit. info.' in README

basilfx committed:
pkg: u8g2: don't toggle unused pins.

Commit:
cpu/sam21_common: remove define of LITTLE_ENDIAN

This define conflicts with LITTLE_ENDIAN defined in
include/machine/endian.h which is part of gcc-arm-none-eabi.

Also, the define does not seem to be used by the ASF included files.

smlng committed:
Merge pull request #5822 from cgundogan/pr/vagrant/add_info_about_vboxusers_group

vagrant: update README and add info about vboxusers group

LudwigKnuepfer committed:
Merge pull request #5696 from LudwigKnuepfer/pr/fix-5695

make: include RIOT_VERSION in CFLAGS for riotbuild

Commit:
boards: add support for SODAQ Autonomo

The following features were tested (briefly):
* UART (the first)
* I2C using a BMP180 on a SODAQ TPH board
* SPI0, connected to the on-board serial data flash (only DevID was read
using the periph_spi test program)
* xtimer is working (it's the same code as in samr21)

boards/sodaq-autonomo: Disable the "big" unittests because it does not fit

mali committed:
drivers/timer: fix copy/paste error

immesys committed:
samd21: i2c: clear timeout_counter

If not cleared, long writes timeout incorrectly

kaspar030 committed:
Merge pull request #5801 from miri64/pkg/enh/lwip-mbox-port

lwip: port mboxes to core_mbox

miri64 committed:
Merge pull request #5831 from mali/timer

drivers/timer: fix copy/paste error

Commit:
samd21: i2c: improve busstate comparison

smlng committed:
kw2xrf: set tx_power in gnrc_netdev_t

without this patch, tx_power is directly set on the device but not in
gnrc_netdev_t. Thus, calling ifconfig in shell shows tx_power always
at 0dBm, never showing the correct, current value. Additionally, it
verifies that given tx_power to be set is in valid range.

mali committed:
boards/arduino-common: use software interrupt for context swap

mali committed:
cpu/atmega_common: fix timer_set_absolute return code

according to timer documentation, timer_set and timer_set_absolute should
return 1 on succes, not 0.

Commit:
Merge pull request #5747 from miri64/tests/feat/od-test

tests: provide tests for od module

Commit:
Makefile.buildtests: Fix typo QUITE -> QUIET

Commit:
drone: Update to Drone v0.5 yaml format

Commit:
travis: Add missing host build group

Commit:
dist/tools/drone-scripts: Delete because Drone can now use the Travis script

Commit:
boards/nucleo-f207: Use cortex_m3_2 MCU_GROUP

Commit:
examples: ccn-lite-relay: minor doc fix

miri64 committed:
Merge pull request #5839 from cgundogan/pr/ccnl_example_docfix

examples: ccn-lite-relay: minor doc fix

miri64 committed:
Merge pull request #5836 from gebart/pr/quiet-typo

Makefile.buildtests: Fix typo QUITE -> QUIET

miri64 committed:
Merge pull request #5832 from immesys/patch-1

samd21: i2c: clear timeout_counter

miri64 committed:
Merge pull request #5827 from miri64/make/enh/quiet-per-default

make: build quiet as default

miri64 committed:
Merge pull request #5766 from zhuoshuguo/fix_cpu_atmega_header_include_guards

fix cpu/atmega header include guardes.

miri64 committed:
Merge pull request #5761 from mali/rmdeadcode

boards/atmega_common: remove dead code on link

smlng committed:
nucleo-f207: fix uart configuration init

aabadie committed:
Merge pull request #5674 from aabadie/driver_io1_xplained

drivers/io1_xplained: initial implementation of Atmel IO1 Xplained extension board

cgundogan committed:
Merge pull request #5843 from smlng/pr/boards/nucleo-f207/fixing_cpp11

nucleo-f207: fix uart configuration init

Commit:
vagrant: sync Tutorials folder into vbox

Commit:
vagrant: prevent error by checking existence of .gitconfig

Commit:
Merge pull request #5794 from smlng/pr/fix_osx_5742

correcting linker LTO flags on OSX

smlng committed:
Merge pull request #5847 from cgundogan/pr/vagrant_file_provisioner

vagrant: prevent error by checking existence of .gitconfig

smlng committed:
Merge pull request #5846 from cgundogan/pr/vagrantfile_tutorial

vagrant: sync Tutorials folder into vbox

smlng committed:
minor typos in README corrected

Commit:
Merge pull request #5850 from smlng/pr/dist_fix_readme

minor typos in README corrected

Commit:
Merge pull request #5838 from gebart/pr/nucleo-f207-mcu-group

boards/nucleo-f207: Use cortex_m3_2 MCU_GROUP

mali committed:
tests: adapt for arduino uno and duemilanove support

- blacklist arduino-uno and arduino-duemilanove for
  coap, libfixmath_unittests, lwip, nhdp,
  pthread, pthread_barrier, pthread_cleanup, pthread_condition_variable
  pthread_cooperation, pthread_rwlock and pthread_tls tests.

- fix sys/pipe build

- unittests: boards added to BOARD_INSUFICIENT_MEMORY list.

Commit:
Merge pull request #5833 from immesys/fix-i2c-checks

samd21: i2c: check all busstate bits

Commit:
vagrant: minimize Vagrantfile and make it loadable

By parameterizing certain paths with `RIOTBASE`, it is possible to load
this Vagrantfile from other Vagrantfiles. This feature will be used by the
Tutorials repository, so that this Vagrantfile has no knowledge about
the tutorials, hence the deletion of the `tutorials` definition at this
place.

smlng committed:
gnrc_rpl: modify RPL default parameters in Makefile

smlng committed:
Merge pull request #5853 from cgundogan/pr/vagrant

vagrant: minimize Vagrantfile and make it loadable

smlng committed:
gnrc_rpl: correct (some) default params

smlng committed:
gnrc_rpl: add explanatory links to RFC sections

smlng committed:
corrects flag compare

miri64 committed:
Merge pull request #5857 from smlng/pr/tests/driver_at86rf2xx/fix

tests/driver_at86rf2xx: corrects flag compare

LudwigKnuepfer committed:
Merge pull request #5763 from jthacker/avr_thread_arch_isr_stack_usage

atmega: use software interrupt for context swap

aabadie committed:
Merge pull request #5451 from mali/uno

boards: add arduino uno and duemilanove support

cgundogan committed:
Merge pull request #5854 from smlng/pr/gnrc_rpl/expose_config_params

gnrc_rpl: adapt default parameters and expose to Makefile

Commit:
Merge pull request #5834 from smlng/pr/drivers/kw2xrf/set_tx_power

kw2xrf: set tx_power in gnrc_netdev_t

kaspar030 committed:
Merge pull request #5743 from keestux/refactor-samd21

Refactor samd21

Commit:
cpu/sam21_common: some corrections to CMSIS samd21 include files

See README.md for details
* eliminate trailing white space
* remove define of LITTLE_ENDIAN

OlegHahm committed:
Merge pull request #5803 from miri64/auto_init_gnrc_netif/fix/mac-prios

auto_init_gnrc_netif: set priorities one lower than 6LoWPAN

Commit:
cpu/sam21_common: support SAMR21E18

Commit:
make: export RIOT_VERSION to CFLAGS_WITH_MACROS

All macros are filtered out of the `CFLAGS` variable and during this
process, the `CFLAGS` string is destroyed. This is due to the fact that
the `RIOT_VERSION` string contains spaces, but the filtering process does
not take the whole string into account (following the first space). This
leads to some parts of the `RIOT_VERSION` string be sill existent in the
`CFLAGS` string.

Instead of exporting the `RIOT_VERSION` string to the `CFLAGS` macro to
just remove it again from there, I exported it to `CFLAGS_WITH_MACROS`.

Fixes #5859

aabadie committed:
Merge pull request #5829 from basilfx/bugfix/u8g2_pins

pkg: u8g2: don't toggle unused pins.

miri64 committed:
ndp: add missing header include

sys/include/net/ndp.h uses IPv6 addresses in NDP message type definitions but up
until now the according header wasn't included.

OlegHahm committed:
Merge pull request #5816 from makomi/shell

sys/shell: fix cppcheck warnings

OlegHahm committed:
Merge pull request #5835 from mali/timer

cpu/atmega_common: fix timer_set_absolute return code

OlegHahm committed:
Merge pull request #5817 from makomi/stm32f0

cpu/stm32f0: uart: suppress cppcheck nullPointer errors

OlegHahm committed:
Merge pull request #5818 from makomi/stm32f3

cpu/stm32f3: uart: suppress cppcheck nullPointer errors

OlegHahm committed:
Merge pull request #5819 from makomi/stm32l1

cpu/stm32l1: uart: suppress cppcheck nullPointer errors

OlegHahm committed:
Merge pull request #5864 from miri64/ndp/fix/incl-header

ndp: add missing header include

Commit:
Merge pull request #5862 from cgundogan/pr/makefile_riot_version

make: export RIOT_VERSION to CFLAGS_WITH_MACROS

PeterKietzmann committed:
Merge pull request #5804 from aeneby/no_mutex_lock

cpu/cc2538: Remove superfluous mutex lock

cgundogan committed:
Merge pull request #5797 from miri64/dist/doc/profile

dist: document IOTLAB_PROFILE variable fo IoT-Lab-support

immesys committed:
samd21: spi: fix power regression

The MISO pin cannot be left floating as it increases power consumption

smlng committed:
cc2538: disable l2 ack recv irq

miri64 committed:
pkg: lwip: do not re-set values already set in netif_add

lwip_netdev2_init() is called by netif_add, which already sets these
values to an according value.

miri64 committed:
pkg: lwip: remove timeout timer

In case of no timeout this timer is not removed, leading to the
re-addition in a later call. This fix removes it properly.

PeterKietzmann committed:
Merge pull request #5869 from smlng/pr/cc2538/disable_accept_l2ack

cc2538: disable l2 ack recv irq

miri64 committed:
Merge pull request #5690 from BytesGalore/hmac_sha256_change_msg_param_type

hashes/sha256: changed data-pointer parameters to `void *`

miri64 committed:
Merge pull request #5691 from BytesGalore/sha1_change_parameter_type

hashes/sha1: changed data-parameter pointers to `void*`

cgundogan committed:
Merge pull request #5837 from gebart/pr/drone-0.5

drone: Update YAML file to Drone 0.5 format

miri64 committed:
Merge pull request #5692 from BytesGalore/md5_change_parameter_type

hashes/md5: changed data-parameter types to `void*`

Lotterleben committed:
unittests: document make debug

Lotterleben committed:
unittests: fix link

miri64 committed:
Merge pull request #5872 from Lotterleben/unittest_gdb_doc

unittests: document make debug

miri64 committed:
Merge pull request #5873 from Lotterleben/unittest_link_fix

unittests: fix link

miri64 committed:
Merge pull request #5381 from cgundogan/pr/netstats_rpl

gnrc_rpl: introduce control plane traffic statistics

PeterKietzmann committed:
Merge pull request #5868 from immesys/patch-1

samd21: spi: fix power regression

kaspar030 committed:
Merge pull request #5758 from miri64/sock/api/initial

sock: Introduction of new application layer API

OlegHahm committed:
debug: add missing line breaks

miri64 committed:
Merge pull request #5686 from OTAkeys/pr/fix_uart_test

tests/periph_uart: include uart_stdio.h

miri64 committed:
Merge pull request #5815 from makomi/periph_gpio

tests/periph_gpio: fix variableScope (cppcheck)

miri64 committed:
Merge pull request #5694 from miri64/gnrc/enh/0-length-payload

gnrc_pktbuf: allow for 0-sized payload snips

PeterKietzmann committed:
Merge pull request #5861 from hamilton-mote/fix-samr21e

cpu/sam21_common: support SAMR21E18

miri64 committed:
sock: doc fixes

miri64 committed:
Merge pull request #5880 from OlegHahm/fix_debug_newlines

debug: add missing line breaks

miri64 committed:
Merge pull request #5493 from OlegHahm/silent_valgrind

native: let valgrind print the leak summary at the end

haukepetersen committed:
boards: removed some superfluous definitions

- removed UART_STDIO_DEV as they use the default value
- removed BTN_B1_PIN for nucleo-207, as its already defined in the common header

kaspar030 committed:
Merge pull request #5890 from haukepetersen/fix_boards_uartstdiodev

boards: removed some superfluous definitions

PeterKietzmann committed:
Merge pull request #5706 from aabadie/nucleo_f446

boards: initial support of ST Nucleo f446

Commit:
kinetis: Add stddef, stdint missing includes for gpio.c

aabadie committed:
Merge pull request #5860 from keestux/add-sodaq-autonomo

boards: add sodaq autonomo

jfbortolotti committed:
nRF52 Fix softdevice.hex flashing issue

Commit:
board/sodaq-autonomo: correct openocd configuration (with example command)

jfbortolotti committed:
cpu: cortexm_common: allow overriding of linker script

PeterKietzmann committed:
Merge pull request #5892 from gebart/pr/kinetis-gpio-include

kinetis: Add stddef, stdint missing includes for gpio.c

miri64 committed:
netdev2_ieee802154: remove NETDEV2_IEEE802154_PAN_COMP flag

miri64 committed:
netdev2_ieee802154: cleanup flag definitions

miri64 committed:
ieee802154: remove need for IEEE802154_BCAST flag

miri64 committed:
tests: adapt unittests for removal of user-controlled PAN_COMP

miri64 committed:
tests: adapt tests for changed broadcast settings

jfbortolotti committed:
dist: tools: jlink.sh: use printf for JLINK_*_FLASH to allow use of backslas

aabadie committed:
Merge pull request #5871 from miri64/pkg/fix/lwip-some-fixes

pkg: lwip: some fixes

Commit:
periph/i2c: convert char to uint8_t where applicapable

In general, data transferred through I2C are bytes and thus should have
type uint8_t, not char.

Also convert uint8_t ptrs to void ptrs

DipSwitch committed:
ndp: fix debug in message in gnrc_ndp_internal_send_nbr_sol

miri64 committed:
Merge pull request #5905 from DipSwitch/pr/fix_debug_ndp_internal_send_nbr_sol_2

ndp: fix debug in message in gnrc_ndp_internal_send_nbr_sol

miri64 committed:
Merge pull request #5895 from jfbortolotti/nRF52_linker

cpu: cortexm_common: allow overriding of linker script

miri64 committed:
Merge pull request #5900 from jfbortolotti/dist_tools_jlink

dist: tools: jlink.sh: use printf for JLINK_*_FLASH to allow use of backslash

jfbortolotti committed:
cpu: cortexm_common: Remove useless path when setting default value for LINKER_SCRIPT

Commit:
ciphers: Documented return values of init functions

Commit:
ciphers: Added defines for return values of cipher_init() and [specific cipher]_init() functions
New definitions: CIPHER_INIT_SUCCESS and CIPHER_ERR_BAD_CONTEXT_SIZE

PeterKietzmann committed:
Merge pull request #5897 from miri64/ieee802154/enh/cleanup

ieee802154: clean-up (Intra-PAN behavior + broadcast)

kaspar030 committed:
Merge pull request #5893 from jfbortolotti/nRF52_flashing

nRF52: Fix softdevice.hex flashing issue

aabadie committed:
Merge pull request #5884 from miri64/sock/doc/fix

sock: doc fixes

Commit:
Merge pull request #5908 from mtausig/ciphers_doc

ciphers: Documented return values of init functions

aabadie committed:
Merge pull request #5902 from keestux/autonomo-openocd

board/sodaq-autonomo: correct openocd configuration

haukepetersen committed:
cpu/sam21: removed superfluous gpio_init_sercom

This function was a duplication of the existing gpio_init_mux,
where the exiting function is the more general one, as the mux
inititalization is not only valid for the sercom module...

haukepetersen committed:
cpu/samd21(common): cleaned up pad selection macros

- renamed uart and spi pad selection macros to consistent style
- adapted uart and spi implementation to use new names

haukepetersen committed:
boards/samr21-xpro: adapted uart and spi config

adjusted names for spi and uart pad selections

haukepetersen committed:
boards/sodaq-autonomo: adapted uart and spi config

adjusted names for spi and uart pad selections

haukepetersen committed:
cpu/sam21_common: remove unused defines

kYc0o committed:
Merge pull request #5670 from astralien3000/dev-aversiveplusplus

Added Aversive++ package.

miri64 committed:
sock: define behavior on invalid addresses

Commit:
Merge pull request #5903 from keestux/i2c-char-to-uint8

periph/i2c: convert char to uint8_t where applicapable

PeterKietzmann committed:
Merge pull request #5912 from haukepetersen/fix_sam0_periphcpu

cpu/sam21_common: remove unused defines

kYc0o committed:
Merge pull request #5475 from aabadie/arduino_zero

boards/arduino-zero: add initial support

thomaseichinger committed:
Merge pull request #5851 from hamilton-mote/hamilton-rtt-stdio

sys/rtt_stdio: Support SEGGER RTT for stdin/stdout

aeneby committed:
cpu/cc2538: Wait for transmission to complete before returning from send()

PeterKietzmann committed:
Merge pull request #5910 from haukepetersen/fix_sam0_muxstuff

cpu/sam21: removed superfluous gpio_init_sercom

miri64 committed:
Merge pull request #5913 from miri64/sock/api/invalid-addresses

sock: define behavior on invalid addresses

haukepetersen committed:
boards/arduino-zero: adapted periph_conf

- changed names of pad selection macros according to
  the CPU changes

PeterKietzmann committed:
Merge pull request #5911 from haukepetersen/fix_sam0_padcfg

cpu/samd21: cleaned up pad selection macros

miri64 committed:
sock: make queue_array parameter a pointer

Since sock_tcp_t isn't defined at this moment (only declared) the
compiler is complaining about the use of an array in the parameter list
here.

    sys/include/net/sock/tcp.h:119:32: error: array type has incomplete element type ‘sock_tcp_t {aka struct
sock_tcp}’
                         sock_tcp_t queue_array[], unsigned queue_len,
                                    ^

PeterKietzmann committed:
cpu/msp430: fix spi clk config for msp430f1xxx

miri64 committed:
Merge pull request #5916 from miri64/sock/fix/tcp-queue-array

sock: make queue_array parameter a pointer

Commit:
boards/sodaq-autonomo: use comma on last element of a list too

Lotterleben committed:
gnrc.h: extend documentation

PeterKietzmann committed:
cpu/msp430: remove non-effective code for msp430f2xxx

smlng committed:
Merge pull request #5915 from aeneby/send_wait_v2

cpu/cc2538: Wait for transmission to complete before returning from send()

Lotterleben committed:
doc: update getting started link in README.md

miri64 committed:
Merge pull request #5907 from jfbortolotti/nRF52_linker

cpu: cortexm_common: Remove useless path when setting default value for LINKER_SCRIPT

miri64 committed:
sock: doc fixes

PeterKietzmann committed:
Merge pull request #5922 from keestux/fix_sam0_padcfg_add_commas

boards/sodaq-autonomo: use comma on last element of a list too

kaspar030 committed:
sock: change "no timeout" value from 0 to UINT32_MAX

aabadie committed:
Merge pull request #5924 from miri64/sock/doc/fixes

sock: doc fixes

miri64 committed:
shell: ncache: Fix #5930

miri64 committed:
Merge pull request #5931 from miri64/sc_ipv6_nc/fix/5930

shell: ncache: Fix #5930

miri64 committed:
Merge pull request #5929 from kaspar030/sock_change_no_timeout_value

sock: change "no timeout" value from 0 to UINT32_MAX

miri64 committed:
sock: Some formatting fixes

Follow-up on #5929

Lotterleben committed:
Merge pull request #5933 from miri64/sock/doc/5929-followup

sock: Some formatting fixes

PeterKietzmann committed:
doc/vagrant: remove information about provisioning

PeterKietzmann committed:
doc/vagrant: add information about changing the vagrant box

cgundogan committed:
Merge pull request #5939 from PeterKietzmann/vagrant_adapt_readme

Vagrant adapt readme

PeterKietzmann committed:
Merge pull request #5828 from cgundogan/pr/packer

packer: initial packer configuration to build virtualbox/vagrant boxes

PeterKietzmann committed:
doc/packer: add initial README

cgundogan committed:
Merge pull request #5940 from PeterKietzmann/add_packer_doc

doc/packer: add initial README

Commit:
Merge pull request #5821 from makomi/tunslip

dist/tools/tunslip: fix cppcheck warnings

Commit:
Merge pull request #5727 from kaspar030/fix_panic_compile

cpu: cortexm_common: fix compile warning

miri64 committed:
Merge pull request #5918 from PeterKietzmann/fix_msp430f1_spi_clk

cpu/msp430: fix spi clk config for msp430f1xxx

smlng committed:
Merge pull request #5947 from Lotterleben/fix_starter_link

doc: update getting started link in README.md

miri64 committed:
Merge pull request #5920 from Lotterleben/doc_net

gnrc.h: extend documentation
